src/lazy/expanded/macro_evaluator.rs

```diff
@@ -114,7 +114,7 @@ where
 #[derive(Copy, Clone)]
 pub struct MacroExpr<'top, D: Decoder> {
     kind: MacroExprKind<'top, D>,
-    variable: Option<TemplateVariableReference<'top>>,
+    pub(crate) variable: Option<TemplateVariableReference<'top>>,
```

Comment:
🪧 Macros now remember the variable (if any) that produced them, allowing them in turn to tag the values they produce.